modules/pam_limits/pam_limits.c

```diff
@@ -1309,28 +1309,30 @@ pam_sm_open_session (pam_handle_t *pamh, int flags UNUSED,
 
     /* Read subsequent *.conf files, if they exist. */
     char **filename_list = read_limits_dir(pamh);
+    char *last_filename = NULL;
     if (filename_list != NULL) {
         for (i = 0; filename_list[i] != NULL; i++) {
             pl->conf_file = filename_list[i];
             retval = parse_config_file(pamh, pwd->pw_name, pwd->pw_uid, pwd->pw_gid, ctrl, pl, 0);
             if (retval != PAM_SUCCESS)
                 break;
         }
+        strcpy(last_filename, pl->conf_file);
         for (i = 0; filename_list[i] != NULL; i++)
             free(filename_list[i]);
         free(filename_list);
     }
 
     if (retval == PAM_IGNORE) {
-        D(("the configuration file ('%s') has an applicable '<domain> -' entry", pl->conf_file));
+        D(("the configuration file ('%s') has an applicable '<domain> -' entry", last_filename));
```

Comment:
`pl->conf_file` is valid here, so no use-after-free happens. Reason: If retval is not PAM_SUCCESS, `break` is called in line 1318, so no free occurred.